pkg/finder/podresourcefinder.go

```diff
@@ -0,0 +1,480 @@
+		return nil, fmt.Errorf("failed to connect: %v", err)
+	}
+
+	finderInstance.client = pb.NewRuntimeServiceClient(finderInstance.conn)
+*/
+	finderInstance.podResourceClient, _, err = podresources.GetClient(finderInstance.args.PodResourceSocketPath, defaultPodResourcesTimeout, defaultPodResourcesMaxSize)
+	if err != nil {
+		return nil, fmt.Errorf("Can't create client: %v", err)
+	}
+	log.Printf("connected to '%v'!", finderInstance.args.PodResourceSocketPath)
+	if finderInstance.args.Namespace != "" {
+		log.Printf("watching namespace %q", finderInstance.args.Namespace)
+	} else {
+		log.Printf("watching all namespaces")
+	}
+
+	return finderInstance, nil
+}
+
+// func getAddressAndDialer(endpoint string) (string, func(ctx context.Context, addr string) (net.Conn, error), error) {
+// 	return util.GetAddressAndDialer(endpoint)
+// }
+//
+// func (f *criFinder) listContainersResponse() (*pb.ListContainersResponse, error) {
+// 	st := &pb.ContainerStateValue{}
+// 	st.State = pb.ContainerState_CONTAINER_RUNNING
+// 	filter := &pb.ContainerFilter{}
+// 	filter.State = st
+//
+// 	ListContReq := &pb.ListContainersRequest{
+// 		Filter: filter,
+// 	}
+//
+// 	ListContResponse, err := f.client.ListContainers(context.Background(), ListContReq)
+// 	if err != nil {
+// 		fmt.Errorf("Error in  ListContResponse: %v", err)
+// 		return nil, err
+// 	}
+// 	return ListContResponse, nil
+// }
+//
+// func (f *criFinder) containerStatsResponse(c *pb.Container) (*pb.ContainerStatsResponse, error) {
+// 	//ContainerStats
+// 	ContStatsReq := &pb.ContainerStatsRequest{
+// 		ContainerId: c.Id,
+// 	}
+// 	ContStatsResp, err := f.client.ContainerStats(context.Background(), ContStatsReq)
+// 	if err != nil {
+// 		log.Printf("Error in  ContStatsResp: %v", err)
+// 		return nil, err
+// 	}
+// 	return ContStatsResp, nil
+// }
+//
+// func (f *criFinder) containerStatusResponse(c *pb.Container) (*pb.ContainerStatusResponse, error) {
+// 	//ContainerStatus
+// 	ContStatusReq := &pb.ContainerStatusRequest{
+// 		ContainerId: c.Id,
+// 		Verbose:     true,
+// 	}
+// 	ContStatusResp, err := f.client.ContainerStatus(context.Background(), ContStatusReq)
+// 	if err != nil {
+// 		log.Printf("Error in  ContStatusResp: %v", err)
+// 		return nil, err
+// 	}
+// 	return ContStatusResp, nil
+// }
+
+func (cpf *PodResourceFinder) updateNUMAMap(numaData map[int]map[v1.ResourceName]int64, ri ResourceInfo) {
+	if ri.Name == v1.ResourceCPU {
+		for _, cpuIDStr := range ri.Data {
+			cpuID, err := strconv.Atoi(cpuIDStr)
+			if err != nil {
+				log.Printf("cannot convert cpuID: %q", cpuIDStr)
+				continue
+			}
+			nodeNum, ok := cpf.cpuID2NUMAID[cpuID]
+			if !ok {
+				log.Printf("unknown cpuID: %d", cpuID)
+				continue
+			}
+			numaData[nodeNum][ri.Name]--
+		}
+		return
+	}
+	for _, pciAddr := range ri.Data {
+		nodeNum, ok := cpf.pciAddr2NUMAID[pciAddr]
+		if !ok {
+			log.Printf("unknown PCI address: %q", pciAddr)
+			continue
+		}
+		numaData[nodeNum][ri.Name]--
+	}
+}
```

Comment:
this is the same code we have in the CRI counterpart, right? can we factor it out somehow?

Reply:
Same as above